make lint does not include inttest

### Before creating an issue, make sure you've checked the following:

- [X] You are running the latest released version of k0s
- [X] Make sure you've searched for existing issues, both open and closed
- [X] Make sure you've searched for PRs too, a fix might've been merged already
- [X] You're looking at docs for the released version, "main" branch docs are usually ahead of released versions.

### Issue

I  believe we can all agree without much discussion that it's a good idea to lint inttest and that it could help preventing some bugs in the tests themselves.

I checked the linting and I don't think anyone would argue about most stuff but, but there are 2 lint errors I think will need some discussion:

```
inttest/common/footloosesuite.go:323:53: context-as-argument: context.Context should be the first parameter of a function (revive)
func (s *FootlooseSuite) cleanupSuite(t *testing.T, ctx context.Context) {
                                                    ^
inttest/common/footloosesuite.go:401:53: context-as-argument: context.Context should be the first parameter of a function (revive)
func (s *FootlooseSuite) dumpNodeLogs(t *testing.T, ctx context.Context, node, dir string) {
```

If we want linting there are two options here:
1- We make an exception and remove the argument `context-as-argument` for inttest
2- We change the function so that context.Context is the first argument,

The convention is [**MOST** functions that use a Context **SHOULD** accept it as their first parameter](https://github.com/golang/go/wiki/CodeReviewComments#contextsl) (emphasis is mine). 
This makes sense to me to some extent but at the same time I think testing.T should go before the Context. Besides the convention says most and should so I understand it's a bit open to interpretation.

Anyway I don't feel very strongly about either option.

Of course we can keep without linting the tests but honestly I think it would be a mistake.

--- a/.golangci.yml
+++ b/.golangci.yml
@@ -5,6 +5,11 @@ run:
   build-tags:
     - hack
   skip-dirs-use-default: false
+  skip-dirs:
+    - build
+    - docs
+    - embedded-bins
+    - examples
   skip-files:
     - "zz_*"
   tests: true

--- a/Makefile
+++ b/Makefile
@@ -19,7 +19,7 @@ endif
 K0S_GO_BUILD_CACHE ?= build/cache
 
 GO_SRCS := $(shell find . -type f -name '*.go' -not -path './$(K0S_GO_BUILD_CACHE)/*' -not -path './inttest/*' -not -name '*_test.go' -not -name 'zz_generated*')
-GO_DIRS := . ./cmd/... ./pkg/... ./internal/... ./static/... ./hack/...
+GO_CHECK_UNIT_DIRS := . ./cmd/... ./pkg/... ./internal/... ./static/... ./hack/...
 
 # EMBEDDED_BINS_BUILDMODE can be either:
 #   docker	builds the binaries in docker
@@ -210,12 +210,18 @@ codegen: $(codegen_targets)
 .PHONY: bindata
 bindata: static/zz_generated_assets.go pkg/assets/zz_generated_offsets_$(TARGET_OS).go
 
-.PHONY: lint
-lint: GOLANGCI_LINT_FLAGS ?=
-lint: .k0sbuild.docker-image.k0s go.sum codegen
+.PHONY: lint-copyright
+lint-copyright:
 	hack/copyright.sh
+
+.PHONY: lint-go
+lint-go: GOLANGCI_LINT_FLAGS ?=
+lint-go: .k0sbuild.docker-image.k0s go.sum codegen
 	CGO_ENABLED=0 $(GO) install github.com/golangci/golangci-lint/cmd/golangci-lint@v$(golangci-lint_version)
-	$(GO_ENV) golangci-lint run --verbose $(GOLANGCI_LINT_FLAGS) $(GO_DIRS)
+	$(GO_ENV) golangci-lint run --verbose $(GOLANGCI_LINT_FLAGS) $(GO_LINT_DIRS)
+
+.PHONY: lint
+lint: lint-copyright lint-go
 
 airgap-images.txt: k0s .k0sbuild.docker-image.k0s 
 	$(GO_ENV) ./k0s airgap list-images --all > '$@' || { \
@@ -254,7 +260,7 @@ smoketests: $(smoketests)
 .PHONY: check-unit
 check-unit: GO_TEST_RACE ?= -race
 check-unit: go.sum codegen
-	$(GO) test -tags=hack $(GO_TEST_RACE) -ldflags='$(LD_FLAGS)' `$(GO) list -tags=hack $(GO_DIRS)`
+	$(GO) test -tags=hack $(GO_TEST_RACE) -ldflags='$(LD_FLAGS)' `$(GO) list -tags=hack $(GO_CHECK_UNIT_DIRS)`
 
 .PHONY: check-image-validity
 check-image-validity: go.sum

--- a/inttest/airgap/airgap_test.go
+++ b/inttest/airgap/airgap_test.go
@@ -51,12 +51,13 @@ func (s *AirgapSuite) TearDownSuite() {
 }
 
 func (s *AirgapSuite) TestK0sGetsUp() {
-	(&common.Airgap{
+	err := (&common.Airgap{
 		SSH:  s.SSH,
 		Logf: s.T().Logf,
 	}).LockdownMachines(s.Context(),
 		s.ControllerNode(0), s.WorkerNode(0),
 	)
+	s.Require().NoError(err)
 
 	s.PutFile(s.ControllerNode(0), "/tmp/k0s.yaml", k0sConfig)
 	s.NoError(s.InitController(0, "--config=/tmp/k0s.yaml"))

--- a/inttest/ap-airgap/airgap_test.go
+++ b/inttest/ap-airgap/airgap_test.go
@@ -74,12 +74,13 @@ func (s *airgapSuite) SetupTest() {
 }
 
 func (s *airgapSuite) TestApply() {
-	(&common.Airgap{
+	err := (&common.Airgap{
 		SSH:  s.SSH,
 		Logf: s.T().Logf,
 	}).LockdownMachines(s.Context(),
 		s.ControllerNode(0), s.WorkerNode(0),
 	)
+	s.Require().NoError(err)
 
 	planTemplate := `
 apiVersion: autopilot.k0sproject.io/v1beta2

--- a/inttest/capitalhostnames/capitalhostnames_test.go
+++ b/inttest/capitalhostnames/capitalhostnames_test.go
@@ -5,7 +5,7 @@ Licensed under the Apache License, Version 2.0 (the "License");
 you may not use this file except in compliance with the License.
 You may obtain a copy of the License at
 
-	http://www.apache.org/licenses/LICENSE-2.0
+    http://www.apache.org/licenses/LICENSE-2.0
 
 Unless required by applicable law or agreed to in writing, software
 distributed under the License is distributed on an "AS IS" BASIS,
@@ -13,6 +13,7 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 */
+
 package basic
 
 import (

--- a/inttest/cnichange/cnichange_test.go
+++ b/inttest/cnichange/cnichange_test.go
@@ -44,11 +44,12 @@ func (s *CNIChangeSuite) TestK0sGetsUpButRejectsToChangeCNI() {
 	s.Require().NoError(err)
 
 	// Wait till we see kube-router DS created
-	watch.DaemonSets(kc.AppsV1().DaemonSets("kube-system")).
+	err = watch.DaemonSets(kc.AppsV1().DaemonSets("kube-system")).
 		WithObjectName("kube-router").
 		Until(s.Context(), func(ds *appsv1.DaemonSet) (bool, error) {
 			return true, nil
 		})
+	s.Require().NoError(err)
 
 	// Restart the controller with new config, should keep kube-router still in use
 	sshC1, err := s.SSH(s.Context(), s.ControllerNode(0))
@@ -60,14 +61,15 @@ func (s *CNIChangeSuite) TestK0sGetsUpButRejectsToChangeCNI() {
 	s.T().Log("restarting k0s")
 	_, err = sshC1.ExecWithOutput(s.Context(), "rc-service k0scontroller restart")
 	s.Require().NoError(err)
-	s.WaitForKubeAPI(s.ControllerNode(0))
+	s.Require().NoError(s.WaitForKubeAPI(s.ControllerNode(0)))
 
 	// check that we see the expeted warning event
-	watch.Events(kc.CoreV1().Events("kube-system")).
+	err = watch.Events(kc.CoreV1().Events("kube-system")).
 		WithFieldSelector(fields.ParseSelectorOrDie("involvedObject.name=k0s")).
 		Until(s.Context(), func(e *corev1.Event) (bool, error) {
 			return e.Type == "Warning" && strings.Contains(e.Message, "cannot change CNI provider from kuberouter to calico"), nil
 		})
+	s.Require().NoError(err)
 }
 
 func TestCNIChangeSuite(t *testing.T) {

--- a/inttest/common/footloosesuite.go
+++ b/inttest/common/footloosesuite.go
@@ -49,7 +49,6 @@ import (
 
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/apimachinery/pkg/util/wait"
 	"k8s.io/client-go/kubernetes"
@@ -201,7 +200,7 @@ func (s *FootlooseSuite) SetupSuite() {
 		// Get a fresh context for the cleanup tasks.
 		ctx, cancel := signalAwareCtx(context.Background())
 		defer cancel(nil)
-		s.cleanupSuite(t, ctx)
+		s.cleanupSuite(ctx, t)
 	}()
 
 	s.waitForSSH(ctx)
@@ -321,8 +320,7 @@ func (s *FootlooseSuite) TearDownSuite() {
 
 // cleanupSuite does the cleanup work, namely destroy the footloose machines.
 // Intended to be called after the suite's context has been canceled.
-func (s *FootlooseSuite) cleanupSuite(t *testing.T, ctx context.Context) {
-
+func (s *FootlooseSuite) cleanupSuite(ctx context.Context, t *testing.T) {
 	if t.Failed() {
 		var wg sync.WaitGroup
 		tmpDir := os.TempDir()
@@ -350,7 +348,7 @@ func (s *FootlooseSuite) cleanupSuite(t *testing.T, ctx context.Context) {
 			wg.Add(1)
 			go func() {
 				defer wg.Done()
-				s.dumpNodeLogs(t, ctx, node, tmpDir)
+				s.dumpNodeLogs(ctx, t, node, tmpDir)
 			}()
 		}
 		wg.Wait()
@@ -399,7 +397,7 @@ func (s *FootlooseSuite) collectTroubleshootSupportBundle(ctx context.Context, t
 	t.Logf("Collected troubleshoot support bundle on %s into %s", node, filePath)
 }
 
-func (s *FootlooseSuite) dumpNodeLogs(t *testing.T, ctx context.Context, node, dir string) {
+func (s *FootlooseSuite) dumpNodeLogs(ctx context.Context, t *testing.T, node, dir string) {
 	ssh, err := s.SSH(ctx, node)
 	if err != nil {
 		t.Logf("Failed to ssh into %s to get logs: %s", node, err.Error())
@@ -1430,7 +1428,7 @@ func (s *FootlooseSuite) GetUpdateServerIPAddress() string {
 }
 
 func (s *FootlooseSuite) AssertSomeKubeSystemPods(client *kubernetes.Clientset) bool {
-	if pods, err := client.CoreV1().Pods("kube-system").List(s.Context(), v1.ListOptions{
+	if pods, err := client.CoreV1().Pods("kube-system").List(s.Context(), metav1.ListOptions{
 		Limit: 100,
 	}); s.NoError(err) {
 		s.T().Logf("Found %d pods in kube-system", len(pods.Items))

--- a/inttest/common/util_test.go
+++ b/inttest/common/util_test.go
@@ -5,7 +5,7 @@ Licensed under the Apache License, Version 2.0 (the "License");
 you may not use this file except in compliance with the License.
 You may obtain a copy of the License at
 
-	http://www.apache.org/licenses/LICENSE-2.0
+    http://www.apache.org/licenses/LICENSE-2.0
 
 Unless required by applicable law or agreed to in writing, software
 distributed under the License is distributed on an "AS IS" BASIS,
@@ -13,6 +13,7 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 */
+
 package common
 
 import (

--- a/inttest/containerdimports/containerd_imports_test.go
+++ b/inttest/containerdimports/containerd_imports_test.go
@@ -115,7 +115,7 @@ func (s *ContainerDImportsSuite) addContainerDRuntime() {
 	defer workerSSH.Disconnect()
 
 	// Make an "alias" runtime using runc
-	workerSSH.Exec(ctx, "ln -s /var/lib/k0s/bin/runc /var/lib/k0s/bin/runfoo", common.SSHStreams{})
+	s.Require().NoError(workerSSH.Exec(ctx, "ln -s /var/lib/k0s/bin/runc /var/lib/k0s/bin/runfoo", common.SSHStreams{}))
 
 	// Configure containerd to use it
 	s.PutFile(s.WorkerNode(0), "/etc/k0s/containerd.d/foo.toml", fooRuntimeConfig)

--- a/inttest/customca/customca_test.go
+++ b/inttest/customca/customca_test.go
@@ -36,7 +36,7 @@ func (s *CustomCASuite) TestK0sGetsUp() {
 	s.Require().NoError(err)
 	defer ssh.Disconnect()
 
-	ssh.Exec(s.Context(), "sh -e", common.SSHStreams{
+	err = ssh.Exec(s.Context(), "sh -e", common.SSHStreams{
 		In: strings.NewReader(fmt.Sprintf(`
 			K0S_PATH=%q
 			IP_ADDRESS=%q
@@ -47,6 +47,7 @@ func (s *CustomCASuite) TestK0sGetsUp() {
 			"$K0S_PATH" token pre-shared --cert /var/lib/k0s/pki/ca.crt --url https://"$IP_ADDRESS":6443/ --out /var/lib/k0s/manifests/test
 		`, s.K0sFullPath, s.GetControllerIPAddress(0))),
 	})
+	s.Require().NoError(err)
 
 	cert, err := ssh.ExecWithOutput(s.Context(), "cat /var/lib/k0s/pki/ca.crt")
 	s.Require().NoError(err)

--- a/inttest/kubeletcertrotate/kubeletcertrotate_test.go
+++ b/inttest/kubeletcertrotate/kubeletcertrotate_test.go
@@ -87,7 +87,8 @@ func (s *kubeletCertRotateSuite) SetupTest() {
 	workerSSH, err := s.SSH(s.Context(), s.WorkerNode(0))
 	s.Require().NoError(err)
 	s.T().Log("waiting to see kubelet rotating the client cert before triggering Plan creation")
-	workerSSH.ExecWithOutput(s.Context(), "inotifywait --no-dereference /var/lib/k0s/kubelet/pki/kubelet-client-current.pem")
+	_, err = workerSSH.ExecWithOutput(s.Context(), "inotifywait --no-dereference /var/lib/k0s/kubelet/pki/kubelet-client-current.pem")
+	s.Require().NoError(err)
 	output, err := workerSSH.ExecWithOutput(s.Context(), "k0s status -ojson")
 	s.Require().NoError(err)
 	status := statusJSON{}

--- a/inttest/toolsuite/operations/planapply.go
+++ b/inttest/toolsuite/operations/planapply.go
@@ -1,16 +1,18 @@
-// Copyright 2022 k0s authors
-//
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-//	http://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
+/*
+Copyright 2022 k0s authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
 
 package operations
 

--- a/inttest/toolsuite/operations/sonobuoy.go
+++ b/inttest/toolsuite/operations/sonobuoy.go
@@ -1,16 +1,18 @@
-// Copyright 2022 k0s authors
-//
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-//	http://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
+/*
+Copyright 2022 k0s authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
 
 package operations
 
@@ -81,8 +83,8 @@ func downloadSonobuoy(ctx context.Context, version string, osname string, arch s
 		}
 	}
 
-	sonobuoyUrl := fmt.Sprintf("https://github.com/vmware-tanzu/sonobuoy/releases/download/v%s/sonobuoy_%s_%s_%s.tar.gz", version, version, osname, arch)
-	resp, err := grab.Get("/tmp", sonobuoyUrl)
+	sonobuoyURL := fmt.Sprintf("https://github.com/vmware-tanzu/sonobuoy/releases/download/v%s/sonobuoy_%s_%s_%s.tar.gz", version, version, osname, arch)
+	resp, err := grab.Get("/tmp", sonobuoyURL)
 	if err != nil {
 		return "", cleanup, fmt.Errorf("failed to download the sonobuoy distribution: %w", err)
 	}

--- a/inttest/toolsuite/tests/conformance_test.go
+++ b/inttest/toolsuite/tests/conformance_test.go
@@ -1,16 +1,18 @@
-// Copyright 2022 k0s authors
-//
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-//	http://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
+/*
+Copyright 2022 k0s authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
 
 package tests
 

--- a/inttest/toolsuite/tests/versionupdate_test.go
+++ b/inttest/toolsuite/tests/versionupdate_test.go
@@ -4,7 +4,7 @@
 // you may not use this file except in compliance with the License.
 // You may obtain a copy of the License at
 //
-//	http://www.apache.org/licenses/LICENSE-2.0
+//     http://www.apache.org/licenses/LICENSE-2.0
 //
 // Unless required by applicable law or agreed to in writing, software
 // distributed under the License is distributed on an "AS IS" BASIS,

--- a/inttest/toolsuite/toolsuite.go
+++ b/inttest/toolsuite/toolsuite.go
@@ -1,16 +1,18 @@
-// Copyright 2022 k0s authors
-//
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-//	http://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
+/*
+Copyright 2022 k0s authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
 
 package toolsuite
 

--- a/inttest/toolsuite/util/terraform.go
+++ b/inttest/toolsuite/util/terraform.go
@@ -1,16 +1,18 @@
-// Copyright 2022 k0s authors
-//
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-//	http://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
+/*
+Copyright 2022 k0s authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
 
 package util
 

